biz.aQute.bndlib/src/aQute/bnd/classfile/ConstantPool.java

```diff
@@ -30,7 +33,102 @@
 	public static final int	CONSTANT_Module				= 19;
 	public static final int	CONSTANT_Package			= 20;
 
-	final Object[]			pool;
+	public enum InfoTag {
+
+		Utf8(CONSTANT_Utf8, String.class), //
+		Integer(CONSTANT_Integer, Integer.class), // ;
+		Float(CONSTANT_Float, Float.class), //
+		Long(CONSTANT_Long, Long.class), //
+		Double(CONSTANT_Double, Double.class), //
+		Class(CONSTANT_Class, ClassInfo.class),
+		String(CONSTANT_String, StringInfo.class), //
+		Fieldref(CONSTANT_Fieldref, FieldrefInfo.class), //
+		Methodref(CONSTANT_Methodref, MethodrefInfo.class), //
+		InterfaceMethodRef(CONSTANT_InterfaceMethodref, InterfaceMethodrefInfo.class), //
+		NameAndType(CONSTANT_NameAndType, NameAndTypeInfo.class), //
+		MethodHandle(CONSTANT_MethodHandle, MethodHandleInfo.class), //
+		MethodType(CONSTANT_MethodType, MethodTypeInfo.class),
+		Dynamic(CONSTANT_Dynamic, DynamicInfo.class), //
+		InvokeDynamic(CONSTANT_InvokeDynamic, InvokeDynamicInfo.class), //
+		Module(CONSTANT_Module, ModuleInfo.class), //
+		Package(CONSTANT_Package, PackageInfo.class);
+
+		public int					tag;
```

Comment:
These should be final and also probably not public instead having accessor methods.